### PR TITLE
Add SQLite mixed-type compatibility for _get_table_columns_and_types

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds changes which seek to further address #38. This extends the work found in #50 for greater compatibility with mixed-types in `_get_table_columns_and_types`.

## What is the nature of your change?

- [x] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: cytotable/convert.py [hunk 1]
```diff
@@ -69,8 +71,39 @@ def _get_table_columns_and_types(source: Dict[str, Any]) -> List[Dict[str, str]]
...
+
+    except duckdb.Error as e:
+        # if we see a mismatched type error
+        # run a more nuanced query through sqlite
+        # to handle the mixed types
+        if "Mismatch Type Error" in str(e) and source_type == ".sqlite":
+            arrow_data_tbl = _sqlite_mixed_type_query_to_parquet(
+                source_path=str(source["source_path"]),
+                table_name=str(source["table_name"]),
+                chunk_size=5,
```
Comment (gwaybio) [inline cytotable/convert.py:96]: This chunk size seems small, also, it is hardcoded. Just checking that this is ok.

**Review by gwaybio — APPROVED** [review]
LGTM! One comment

File: cytotable/convert.py (hunk 1 above)
Comment (d33bs) [reply cytotable/convert.py:99]: Thanks for the check on this - `chunk_size` is set to 5 here so as to use a similar SQL `LIMIT` on row results from the table for data type inference. The nomenclature of the parameter stems from how the function is used when extracting "chunks" of data later on. We use this constraint on the dataset in this case so as to enable "good enough" data type inferencing while balancing performance (not reading the full table will increase the speed by which the task completes). I added some code comments to increase the description of what's happening in these areas.

**Comment by d33bs** [comment]
Thank you @gwaybio for the review! Will merge pending no unexpected tests failing.

**Pull request merged**